src/unix/process.c

```diff
@@ -563,14 +575,12 @@ int uv_spawn(uv_loop_t* loop,
 
 error:
   if (pipes != NULL) {
-    for (i = 0; i < stdio_count; i++) {
-      if (i < options->stdio_count)
-        if (options->stdio[i].flags & (UV_INHERIT_FD | UV_INHERIT_STREAM))
-          continue;
-      if (pipes[i][0] != -1)
-        uv__close_nocheckstdio(pipes[i][0]);
-      if (pipes[i][1] != -1)
-        uv__close_nocheckstdio(pipes[i][1]);
+    for (i = 0; i < options->stdio_count; i++) {
+      if ((options->stdio[i].flags & (UV_INHERIT_FD | UV_INHERIT_STREAM)) ||
+          pipes[i][0] == -1)
+        continue;
+      uv__close_nocheckstdio(pipes[i][0]);
+      uv__close_nocheckstdio(pipes[i][1]);
```

Comment:
I'm not sure I follow why this fixes the double close. Can you elaborate?

Reply:
The only `stdio`s where `pipes[i][0] != -1` are ones where `socketpair()` was called, which are the ones we want to close.

Reply:
Thanks, I understand now but can you add a comment explaining that? It's not obvious just from context. Can I suggest an edit for readability while you're here?
```c
if (options->stdio[i].flags & (UV_INHERIT_FD | UV_INHERIT_STREAM))
  continue;
/* <comment goes here> */
if (pipes[i][0] == -1)
  continue;
```

Reply:
Absolutely. Done :heavy_check_mark: